### PR TITLE
Update discovery form to use the new API and machine list box

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- Update discovery form to use the new API and machine list box

Note: includes changes from: https://github.com/canonical/maas-ui/pull/4389 - that needs to be merged first.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to network discovery
- Expand the dropdown on the right hand side and select "Add discovery..."
- Select a machine from the list in the Parent field and
- Ensure the discovery has been saved correctly

## Fixes

Fixes: https://github.com/canonical/app-tribe/issues/1112

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

It could be helpful to provide some screenshots to aid in QAing the change.

### CODE REVIEW

**Comment by huwshimi** [comment]
@petermakowski I'm going to hold off reviewing this one until we're happy with the the other PR.

File: src/app/base/components/DhcpFormFields/DhcpFormFields.tsx [hunk 1]
```diff
@@ -134,21 +131,21 @@ export const DhcpFormFields = ({ editing }: Props): JSX.Element => {
           { value: "device", label: "Device" },
         ]}
       />
-      {type &&
+      {type === "machine" ? (
+        <FormikField component={MachineSelect} name="entity" />
```
Comment (petermakowski) [inline src/app/base/components/DhcpFormFields/DhcpFormFields.tsx:135]: Is this what you had in mind @huwshimi ?

Comment (huwshimi) [reply src/app/base/components/DhcpFormFields/DhcpFormFields.tsx:135]: Yep! And then you need to wrap the contents in a `<Field>` component (here's a good example: https://github.com/canonical/maas-ui/blob/main/src/app/base/components/TagSelector/TagSelector.tsx#L250).

Comment (petermakowski) [reply src/app/base/components/DhcpFormFields/DhcpFormFields.tsx:135]: Done

**Comment by petermakowski** [comment]
> @petermakowski I'm going to hold off reviewing this one until we're happy with the the other PR.

Decided to merge this to the main PR https://github.com/canonical/maas-ui/pull/4389 as this isn't going to get reviewed today anyway and will make it easier for you to carry on working on this if it's all in a single branch.

**Pull request closed without merging**